src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+            return;
+        }
+        func_80AB6100(this, globalCtx, 1);
+    }
+}
+
+void func_80AB7290(EnNiw* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060000E8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060000E8), 0, -10.0f);
+    this->unk_2A0 = Math_Rand_ZeroFloat(1.99f);
+    this->actor.speedXZ = 4.0f;
+    this->actionFunc = func_80AB7328;
+}
+
+void func_80AB7328(EnNiw* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->timer6 == 0) {
+        this->unk_2AC = this->unk_2B8 = this->actor.posRot.pos.x;
+        this->unk_2B0 = this->unk_2BC = this->actor.posRot.pos.y;
+        this->unk_2B4 = this->unk_2C0 = this->actor.posRot.pos.z;
+        this->timer5 = this->timer4 = this->unk_29E = 0;
+        this->unk_26C[7] = this->unk_26C[5] = this->unk_26C[6] = this->unk_26C[8] = this->actor.speedXZ =
+            this->unk_2FC = this->unk_300 = 0.0f;
+
+        if (this->actor.params == 0x0004) {
+            this->actor.params = 0x0000;
+        }
+
+        this->actionFunc = EnNiw_ResetAction;
+    } else {
+        this->unk_2E4 = Math_atan2f(this->actor.posRot.pos.x - player->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.z - player->actor.posRot.pos.z) *
+                        10430.378f;
+        func_80AB6100(this, globalCtx, 0);
+        func_80AB5BF8(this, globalCtx, 2);
+    }
+}
+
+void func_80AB7420(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        this->unk_2A4 = (s16)Math_Rand_ZeroFloat(3.99f) + 5;
+        this->actionFunc = EnNiw_ResetAction;
+    }
+}
+
+void func_80AB747C(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk_2A8 == 0 && this->actor.params != 0x000A && this->actionFunc != func_80AB6450 &&
+        this->collider.base.acFlags & 2) {
+        this->collider.base.acFlags &= 0xFFFD;
+        this->sfxTimer1 = 30;
+        if (this->unk_2A4 > 0) {
+            if (D_80AB85E0 == 0) {
+                this->unk_2A4--;
+            }
+        }
+        this->unk_2A6 = 1;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+        this->timer6 = 100;
+        this->path = 0;
+        this->actionFunc = func_80AB7290;
+    }
+}
+
+void EnNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 pad;
+    EnNiw* this = THIS;
+    Player* player = PLAYER;
+    s16 i;
+    s16 phi_v1;
+    Vec3f spC4 = { 0.0f, 0.0f, 0.0f };
+    Vec3f spB8 = { 0.0f, 0.0f, 0.0f };
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    f32 pad2;
+    f32 sp8C;
+    Vec3f cam;
+    f32 sq;
+    f32 camResult;
+    f32 pad3[10];
+
+    this->unk_294++;
+
+    if (this->actionFunc != func_80AB6570) {
+        this->unk_26C[9] = 0.0f;
+    }
+
+    if (this->unk_2A6) {
+        phi_v1 = 20;
+
+        if (this->unk_2A6 == 2) {
+            phi_v1 = 4;
+        }
+
+        for (i = 0; phi_v1 > i; i++) {
```

Comment:
```suggestion
        for (i = 0; i < phi_v1; i++) {
```
is a way more common way to write it